internal/flipflop/handler.go

```diff
@@ -132,11 +132,18 @@ func (cth *ConditionTaskHandler) Run(ctx context.Context) error {
 	case rctypes.PowerCycleBMC:
 		return cth.powerCycleBMC(ctx)
 	case rctypes.SetPowerState:
-		return cth.setPowerState(ctx)
+		return cth.setPowerState(ctx, cth.task.Parameters.ActionParameter)
 	case rctypes.SetNextBootDevice:
-		return cth.setNextBootDevice(ctx)
+		return cth.setNextBootDevice(
+			ctx,
+			cth.task.Parameters.ActionParameter,
+			cth.task.Parameters.SetNextBootDevicePersistent,
+			cth.task.Parameters.SetNextBootDeviceEFI,
+		)
 	case rctypes.ValidateFirmware:
 		return cth.validateFirmware(ctx)
+	case rctypes.PxeBootPersistent:
```

Comment:
Also, what is the point of this being its own type? PxeBootPersistent is just
setNextBootDevice() with action being "pxe", and persistent being "true". So this is duplicate functionality.

Reply:
The type enables setting PXE boot and powering on the server https://github.com/metal-toolbox/flipflop/pull/14/commits/0a3a368f4fb09b8dea29b229fafbd93a41921378#diff-99e0e19629127313b5abddf246d61aa9be84d00e226d34e35d6e8f4ffe9a0b0aR284-R291

Reply:
Ah ok. But this could still be completed by just having two conditions? Also not very clear with the rctype name that it also powers on the server.

Reply:
It could be two conditions as well, although the whole point of flipflop was to accept and carry out this sort of work. Pushing this responsibility to the Condition Orchestrator will required adding in delay and other pxe/powercycling bits that does not belong there.

If you have another name, do suggest